src/OrleansRuntime/ReminderService/LocalReminderService.cs

```diff
@@ -400,7 +477,7 @@ private bool TryStopPreviousTimer(GrainReference grainRef, string reminderName)
         private async Task DoResponsibilitySanityCheck(GrainReference grainRef, string debugInfo)
         {
             if (status != ReminderServiceStatus.Started)
-                await startedTask.Task;
+                await startedTask.Task.WithTimeout(InitialLoadMaxWaitTimeForUpdates);
```

Comment:
But that could potentially make the grain be stuck for 10 minutes. I would prefer to bail early.